Hakawai/Mentions/HKWMentionsPluginV2.m

```diff
@@ -1052,6 +1059,28 @@ - (void)handleMentionsCreationInText:(NSString *)text atLocation:(NSUInteger)loc
     }
 }
 
+- (void)textView:(UITextView *)textView willPasteTextInRange:(NSRange)range {
+    if (self.currentlySelectedMentionRange.location != NSNotFound) {
+        [self bleachExistingMentionAtRange:self.currentlySelectedMentionRange];
+        self.currentlySelectedMentionRange = NSMakeRange(NSNotFound, 0);
+    } else {
+        NSRange mentionRangeAtStartOfRange;
+        HKWMentionsAttribute *mentionAtStartOfRange = [self mentionAttributeAtLocation:range.location range:&mentionRangeAtStartOfRange];
+        BOOL doesStartOfRangeIntersectWithMention = mentionAtStartOfRange && mentionRangeAtStartOfRange.location != range.location;
+        if (doesStartOfRangeIntersectWithMention) {
+            [self bleachExistingMentionAtRange:mentionRangeAtStartOfRange];
+        }
+
+        NSRange mentionRangeAtEndOfRange;
+        HKWMentionsAttribute *mentionAtEndOfRange = [self mentionAttributePrecedingLocation:range.location+range.length range:&mentionRangeAtEndOfRange];
+        BOOL doesEndOfRangeIntersectWithMention = mentionAtEndOfRange
+        && mentionRangeAtEndOfRange.location + mentionRangeAtEndOfRange.length != range.location + range.length;
+        if (doesEndOfRangeIntersectWithMention) {
+            [self bleachExistingMentionAtRange:mentionRangeAtEndOfRange];
+        }
```

Comment:
can we extract this into a helper and have them shared between this function and `- (BOOL)textView:(UITextView *)textView shouldChangeTextInRange:(NSRange)range replacementText:(NSString *)text {` ?